app/Http/Controllers/UploadController.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+use Illuminate\Http\Request;
+
+class UploadController extends \Controller
+{
+    public function __construct()
+    {
+    	// $this->middleware('auth');
+    }
+
+    public function show($user_id, $filename, Request $request)
+    {
+    	if(Auth::user() && (Auth::user()->id == $user_id || Auth::user()->hasRole('administrator')))
```

Comment:
If the `auth` middleware is re-enabled, you should be able to assume `Auth::user()` is set here.